packages/tx/src/capabilities/legacy.ts

```diff
@@ -256,9 +256,10 @@ export function sign(
   }
 
   const msgHash = tx.getHashedMessageToSign()
-  const ecSignFunction = tx.common.customCrypto?.ecsign ?? ecsign
-  const { v, r, s } = ecSignFunction(msgHash, privateKey, { extraEntropy })
-  const signedTx = tx.addSignature(v, r, s, true)
+  const ecSignFunction = tx.common.customCrypto?.ecsign ?? secp256k1.sign
+  const { recovery, r, s } = ecSignFunction(msgHash, privateKey, { extraEntropy })
+  // TODO: addSignature `v` can likely be converted to type `number` instead `bigint`
```

Comment:
Is there any major reason to do this (or not do it)?

Reply:
This is done in secp256k1.sign output, so we should do here also. No reason not to do it :)